theme/icons_test.go

```diff
@@ -1,36 +1,399 @@
 }
 
-func TestNewThemedResource(t *testing.T) {
+func TestNewThemedResource_TwoStaticResourceSupport(t *testing.T) {
 	fyne.CurrentApp().Settings().SetTheme(DarkTheme())
-	custom := NewThemedResource(mailcomposeDark, mailcomposeLight)
+	custom := NewThemedResource(helperNewStaticResource(), helperNewStaticResource())
 	content := custom.Content()
+	nm := custom.Name()
 
 	fyne.CurrentApp().Settings().SetTheme(LightTheme())
 	assert.NotEqual(t, content, custom.Content())
+	assert.Equal(t, nm, custom.Name())
+}
+
+func TestNewThemedResource_OneStaticResourceSupport(t *testing.T) {
+	fyne.CurrentApp().Settings().SetTheme(DarkTheme())
+	custom := NewThemedResource(helperNewStaticResource(), nil)
+	content := custom.Content()
+	nm := custom.Name()
+
+	fyne.CurrentApp().Settings().SetTheme(LightTheme())
+	assert.NotEqual(t, content, custom.Content())
+	assert.Equal(t, nm, custom.Name())
+}
+
+func TestThemedResource_Name(t *testing.T) {
+	sr := fyne.NewStaticResource("cancel_Paths.svg",
+		helperLoadBytes(t, "cancel_Paths.svg"))
+	dr := &ThemedResource{
+		source: sr,
+	}
+	assert.Equal(t, sr.Name(), dr.Name())
+}
+
+func TestThemedResource_Content_NoGroupsFile(t *testing.T) {
+	fyne.CurrentApp().Settings().SetTheme(DarkTheme())
+	sr := fyne.NewStaticResource("cancel_Paths.svg",
+		helperLoadBytes(t, "cancel_Paths.svg"))
+	dr := &ThemedResource{
+		source: sr,
+	}
+	assert.NotEqual(t, sr.Content(), dr.Content())
+}
+
+func TestThemedResource_Content_GroupPathFile(t *testing.T) {
+	fyne.CurrentApp().Settings().SetTheme(DarkTheme())
+	sr := fyne.NewStaticResource("check_GroupPaths.svg",
+		helperLoadBytes(t, "check_GroupPaths.svg"))
+	dr := &ThemedResource{
+		source: sr,
+	}
+	assert.NotEqual(t, sr.Content(), dr.Content())
+}
+
+func TestThemedResource_Content_GroupRectFile(t *testing.T) {
+	fyne.CurrentApp().Settings().SetTheme(DarkTheme())
+	sr := fyne.NewStaticResource("info_GroupRects.svg",
+		helperLoadBytes(t, "info_GroupRects.svg"))
+	dr := &ThemedResource{
+		source: sr,
+	}
+	assert.NotEqual(t, sr.Content(), dr.Content())
+}
+
+func TestThemedResource_Content_GroupPolygonsFile(t *testing.T) {
+	fyne.CurrentApp().Settings().SetTheme(DarkTheme())
+	sr := fyne.NewStaticResource("warning_GroupPolygons.svg",
+		helperLoadBytes(t, "warning_GroupPolygons.svg"))
+	dr := &ThemedResource{
+		source: sr,
+	}
+	assert.NotEqual(t, sr.Content(), dr.Content())
+}
+
+// a black svg object omits the fill tag, this checks it it still properly updated
+func TestThemedResource_Content_BlackFillIsUpdated(t *testing.T) {
+	fyne.CurrentApp().Settings().SetTheme(DarkTheme())
+	sr := fyne.NewStaticResource("cancel_PathsBlackFill.svg",
+		helperLoadBytes(t, "cancel_PathsBlackFill.svg"))
+	dr := &ThemedResource{
+		source: sr,
+	}
+	assert.NotEqual(t, sr.Content(), dr.Content())
+}
+
+// Test Asset Sources
+func Test_FyneLogo(t *testing.T) {
+	result := FyneLogo().Name()
+	fyne.CurrentApp().Settings().SetTheme(LightTheme())
```

Comment:
These lines also could be removed I think, just like the SetTheme(DarkTheme()) already removed?

Reply:
Oh, I remember why I did those tests that way now.  I was checking to make sure the resource didn't change when the themes changed in addition to verifying that the correct .svg resource was being used.  What do you think? Should we just check for the resource being correct?  Without the theme change I agree it's not useful to set the theme.

Reply:
I probably should have named those tests better too, it doesn't describe exactly what they're testing.

Reply:
I'd say yeah, check the result. The side-effect of testing the theme change as well would be a different test, if required at all. I think that it's possibly not needed.

Reply:
agreed, fixed and pushed